tests/vfs_open_mode_tests.cpp

```diff
@@ -0,0 +1,99 @@
+#include <sqlite_orm/sqlite_orm.h>
+#include <catch2/catch_all.hpp>
+#include <cstdio>  //  std::remove
+#if SQLITE_ORM_HAS_INCLUDE(<filesystem>)
+#include <filesystem>
+#endif
+#include "catch_matchers.h"
+
+#ifdef SQLITE_ORM_CTAD_SUPPORTED
+
+struct User {
+    std::string id;
+};
+
+using namespace sqlite_orm;
+
+static const auto default_table = make_table("users", make_column("id", &User::id, primary_key()));
+
+TEST_CASE("vfs modes open successfully") {
+
+#if defined(SQLITE_ORM_APPLE)
+    internal::string_constant_type vfs =
+        GENERATE(unix_vfs_name, unix_posix_vfs_name, unix_dotfile_vfs_name, unix_afp_vfs_name);
+#elif defined(SQLITE_ORM_UNIX)
+    internal::string_constant_type vfs = GENERATE(unix_vfs_name, unix_posix_vfs_name, unix_dotfile_vfs_name);
+#elif defined(SQLITE_ORM_WIN)
+    internal::string_constant_type vfs = GENERATE(win32_vfs_name, win32_longpath_vfs_name);
+#endif
+
+    connection_control options{true, std::string(vfs)};
+    auto storage = make_storage(":memory:", options, default_table);
+    UNSCOPED_INFO("FAILED VFS: " << vfs);
+    REQUIRE_NOTHROW(storage.open_forever());
+
+    REQUIRE(storage.vfs_name() == vfs);
+    REQUIRE(storage.open_mode() == db_open_mode::default_);
+
+    SECTION("Storage copy operator carries over vfs option") {
+        auto storage_copy = storage;
+        CHECK(storage_copy.is_opened());
+        CHECK(storage_copy.vfs_name() == vfs);
+        CHECK(storage_copy.open_mode() == db_open_mode::default_);
+    }
+}
+
+TEST_CASE("readwrite/readonly open modes behaves as expected") {
+    const bool in_memory = GENERATE(true, false);
+    const char* tmp_filename = in_memory ? ":memory:" : "open_mode.sqlite";
+
+    connection_control options{true}, readonly_options{true};
```

Comment:
please skip declaring >1 vars in one statement - it is hard to read. And as far as I remember the code stile in wiki requires to avoid such statements

Reply:
The assignment from the ternary expression was OK, it should have been `readonly_options` that should be defined on a separate line 😉 

But that's not why the unit tests fail - I am not exactly sure why, however:
* We enabled a compiler error for the wrong order of member initializers:
  > C:\projects\sqlite-orm\include\sqlite_orm\sqlite_orm.h(13941,51): error C5038: data member 'sqlite_orm::internal::connection_holder::filename' will be initialized after data member 'sqlite_orm::internal::connection_holder::vfs_name'
* it's good if you merge again dev -> 1392-support-sqlite-vfs